Realm/RLMQueryUtil.mm

```diff
@@ -663,15 +663,18 @@ void add_collection_operation_constraint(NSPredicateOperatorType operatorType,
 
     if (diacriticSensitive) {
         switch (operatorType) {
-            case NSBeginsWithPredicateOperatorType:
-                add_substring_constraint(std::forward<T>(value), column.begins_with(value, caseSensitive));
+            case NSBeginsWithPredicateOperatorType: {
```

Comment:
Ahhh! I was originally going to construct the queries as temporary variables and then move them into the `add_substring_constraint` call, but then realized that was unnecessary.